daiquiri/datalink/adapter.py

```diff
@@ -0,0 +1,165 @@
+from django.conf import settings
+from django.urls import reverse
+
+from daiquiri.core.constants import ACCESS_LEVEL_PUBLIC
+from daiquiri.core.adapter import DatabaseAdapter
+from daiquiri.core.utils import import_class, get_doi_url
+
+
+def DatalinkAdapter():
+    return import_class(settings.DATALINK_ADAPTER)()
+
+
+class BaseDatalinkAdapter(object):
+    """
+    Each datalink adapter needs to configure a set of resource types.
+
+    For each resource type, the following methods need to be implemented:
+
+    * get_<resource_type>_list(self): returns a list of all resources for the resource_type
```

Comment:
The more important point here is that the list must come from the Database and it should not be build from the model:

```suggestion
    * get_<resource_type>_list(self): returns the list of all resources from the database for a given resource_type
```